src/test/java/io/cdap/plugin/salesforce/etl/SalesforceBatchSinkETLTest.java

```diff
@@ -0,0 +1,252 @@
+        .set("ForecastCategoryName", "Closed")
+        .build()
+    );
+
+    Map<String, String> sinkProperties = new ImmutableMap.Builder<String, String>()
+      .put(SalesforceSinkConfig.PROPERTY_MAX_RECORDS_PER_BATCH, "1").build();
+    List<SObject> createdSObjects = new ArrayList<>();
+
+    ApplicationManager appManager = deployPipeline(sObject, schema);
+    runPipeline(appManager, inputRecords);
+    assertRecordsCreated(sObject, inputRecords, createdSObjects);
+    addToCleanUpList(createdSObjects);
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testCompoundFieldsValidation() throws Exception {
+    String sObject = "Account";
+    Schema schema = Schema.recordOf("output",
+                                    Schema.Field.of("Name", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("NumberOfEmployees", Schema.of(Schema.Type.INT)),
+                                    // no compund fields like 'BillingAddress' are allowed.
+                                    // Schema validation should fail
+                                    Schema.Field.of("BillingAddress", Schema.of(Schema.Type.STRING))
+    );
+
+    // Exception below is expected:
+    // "java.lang.IllegalStateException: Expected 200 OK, got 400 Bad Request. Error: Failed to deploy app"
+    // This indicates that validation (not actual execution) failed.
+    ApplicationManager appManager = deployPipeline(sObject, schema);
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testNonCreatableFieldsValidation() throws Exception {
+    String sObject = "Opportunity";
+    Schema schema = Schema.recordOf("output",
+                                    Schema.Field.of("Name", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("StageName", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("CloseDate", Schema.of(Schema.LogicalType.DATE)),
+                                    Schema.Field.of("StageName", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("IsPrivate", Schema.of(Schema.Type.BOOLEAN)),
+                                    Schema.Field.of("Amount", Schema.of(Schema.Type.DOUBLE)),
+                                    Schema.Field.of("ForecastCategoryName", Schema.of(Schema.Type.STRING)),
+                                    // HasOverdueTask fields is present, but not-creatable in Opportunity sObject
+                                    Schema.Field.of("HasOverdueTask", Schema.of(Schema.Type.STRING))
+    );
+
+    // Exception below is expected:
+    // "java.lang.IllegalStateException: Expected 200 OK, got 400 Bad Request. Error: Failed to deploy app"
+    // This indicates that validation (not actual execution) failed.
+    deployPipeline(sObject, schema);
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testNonExistingFieldsValidation() throws Exception {
```

Comment:
changed that to run the validation method. However this still has to remain in integration tests (not unit tests). Since validation process needs connection to Salesforce instance.